### PR TITLE
providers: add support for scaleway

### PR DESCRIPTION
**Pull request by xinau**
This change adds support for [Scaleway](https://scaleway.com) as a platform by adding a new provider. The new `ScalewayProvider` has the capability to

* Get attributes
* Check-in after boot
* Retrieve SSH keys

### CODE REVIEW

File: docs/usage/attributes.md [hunk 1]
```diff
@@ -115,6 +115,14 @@ Cloud providers with supported metadata endpoints and their respective attribute
 * powervs
   - AFTERBURN_POWERVS_INSTANCE_ID
   - AFTERBURN_POWERVS_LOCAL_HOSTNAME
+* scaleway
+  - AFTERBURN_SCALEWAY_HOSTNAME
+  - AFTERBURN_SCALEWAY_INSTANCE_ID
+  - AFTERBURN_SCALEWAY_INSTANCE_TYPE
+  - AFTERBURN_SCALEWAY_PRIVATE_IPV4
+  - AFTERBURN_SCALEWAY_PUBLIC_IPV4
```
Comment (prestist) [inline docs/usage/attributes.md:123]: Nit: Can we change this to be "AFTERBURN_SCALEWAY_IPV4_PUBLIC" Like most of the other providers do? and then do so for the other similar attributes?

**Review by prestist — COMMENTED** [review]
Wow thank you so much for working on this! Great work.

Also make sure to add a line in the afterburn [release notes doc](https://github.com/coreos/afterburn/blob/main/docs/release-notes.md)


Additionally please go ahead and file a Platform request to ensure all parts are covered. 
https://github.com/coreos/fedora-coreos-tracker/issues/new/choose

**Comment by xinau** [comment]
@prestist Thanks for the review. Sorry for the delayed response, been on vacation :). Going to resolve you're comments later today/tomorrow.

**Comment by miabbott** [comment]
> Additionally please go ahead and file a Platform request to ensure all parts are covered.
> https://github.com/coreos/fedora-coreos-tracker/issues/new/choose

Found an existing platform request for Scaleway - https://github.com/coreos/fedora-coreos-tracker/issues/866

**Comment by xinau** [comment]
@prestist Could you give this another review? Anything else that needs to be adjusted?
@miabbott Thanks for doing the due diligence of searching for the open platform request.

**Comment by prestist** [comment]
@xinau Sorry for the silence; taking a look today and will get back to you if I have any other questions :)

**Review by prestist — COMMENTED** [review]
Alright everything looks great! thank you :) 

The only nit that I would want fixed is if you could join the 3 commits into 1? 

Thank you!

**Comment by xinau** [comment]
Going to join the commits this evening. Thanks for the assistance :)

**Comment by xinau** [comment]
@prestist I've rebased the branch and put the 3 commits into a single one. Sorry it took a bit longer than expected.

**Review by prestist — APPROVED** [review]
LGTM, thank you again for being so diligent on this and adding support for a new provider!

**Comment by xinau** [comment]
@prestist Thanks for your support

**Pull request merged**